{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "interesting-curve",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import random\n",
    "import numpy as np\n",
    "import scanpy as sc\n",
    "# import scipy\n",
    "from tqdm import tqdm\n",
    "\n",
    "data_path = '/data2/hratch/immune_CCI/covid/covid_atlas/'\n",
    "load_h5 = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "higher-merit",
   "metadata": {},
   "outputs": [],
   "source": [
    "pbmc_covid = sc.read_mtx(data_path + 'raw/GSE158055_covid19_counts.mtx.gz') # raw counts\n",
    "if load_h5: \n",
    "    pch5 = sc.read_h5ad(data_path + 'raw/COVID19_ALL.h5ad') # load dataset\n",
    "# pbmc_covid = sc.read_10x_mtx(data_path + 'raw/counts/')\n",
    "\n",
    "print('Finished loading covid datasets')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "recovered-lease",
   "metadata": {},
   "outputs": [],
   "source": [
    "# exclude samples with fewer than 2000 cells\n",
    "md_cell = pd.read_csv(data_path + 'raw/GSE158055_cell_annotation.csv.gz')\n",
    "n_samples = md_cell.sampleID.value_counts()\n",
    "samples_to_keep = n_samples[n_samples > 2000].index.tolist()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "refined-trust",
   "metadata": {},
   "outputs": [],
   "source": [
    "md = pd.read_excel(data_path + 'raw/GSE158055_sample_metadata.xlsx', sheet_name = 0, skiprows=20)\n",
    "md = md.iloc[range(304 - 20), range(25)]\n",
    "md = md[md['Sample name'].isin(samples_to_keep)]\n",
    "\n",
    "\n",
    "contexts = md['characteristics: CoVID-19 severity'].unique()\n",
    "n_contexts = contexts.shape[0]\n",
    "\n",
    "context_counts = md['characteristics: CoVID-19 severity'].value_counts() \n",
    "min_context_type = context_counts[context_counts == context_counts.min()].index.tolist()[0]\n",
    "min_context_count = len(md[md['characteristics: CoVID-19 severity'] == min_context_type]['Patients'].unique())\n",
    "\n",
    "max_samples = min_context_count*n_contexts\n",
    "\n",
    "context_map = {context: md[md['characteristics: CoVID-19 severity'] == \\\n",
    "                           context][['Sample name', 'Patients']].reset_index(drop = True) for context in contexts}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "painted-assistant",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/pandas/core/internals/blocks.py:993: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.\n",
      "  arr_value = np.array(value)\n"
     ]
    }
   ],
   "source": [
    "# randomly select samples subsetted from the entire dataset\n",
    "# make sure to choose an even number of each context\n",
    "# make sure not to repeat patients within a context\n",
    "\n",
    "n_iter = 1 # number of times to run subsetting\n",
    "seed = 0\n",
    "\n",
    "Samples = pd.DataFrame(columns = ['iteration', 'n_samples', 'sample_names'])\n",
    "idx = 0\n",
    "\n",
    "sample_iters = [3, 6, 12, 24, 36, 48, 60, 75]#list(range(n_contexts, max_samples + 1, n_contexts))\n",
    "\n",
    "for iteration in range(n_iter):\n",
    "    for n_samples in sample_iters:\n",
    "        cmap_temp = context_map.copy()\n",
    "        n_sample_per_context = int(n_samples/n_contexts)\n",
    "        samples = list()\n",
    "        for context in contexts:\n",
    "            df = context_map[context].sample(frac=1, random_state = seed).drop_duplicates(subset = 'Patients') # shuffle rows to randomly drop duplicates\n",
    "            samples += df.sample(n = n_sample_per_context, random_state = seed)['Sample name'].tolist()\n",
    "            seed += 1\n",
    "        Samples.loc[idx,:] = [iteration, n_samples, samples]\n",
    "        idx += 1\n",
    "\n",
    "Samples['sample_names'] = Samples.sample_names.apply(lambda x: '; '.join(x))            \n",
    "Samples.to_csv(data_path + 'interim/timing_inputs/samples_for_timing.csv')            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "intimate-digest",
   "metadata": {},
   "outputs": [],
   "source": [
    "#cell_ids = pd.read_csv(data_path + 'raw/GSE158055_covid19_barcodes.tsv.gz', header = None)\n",
    "gene_ids = pd.read_csv(data_path + 'raw/GSE158055_covid19_features.tsv.gz', header = None)\n",
    "md_cell.set_index('cellName', drop = True, inplace = True)\n",
    "\n",
    "pbmc_covid = pbmc_covid.transpose() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "intellectual-serum",
   "metadata": {},
   "outputs": [],
   "source": [
    "if load_h5:\n",
    "    pbmc_covid.obs = pch5.obs\n",
    "    pbmc_covid.var = pch5.var\n",
    "else:\n",
    "    pbmc_covid.obs = md_cell\n",
    "    pbmc_covid.var = gene_ids.set_index(0, drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "nasty-desktop",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|          | 0/140 [00:00<?, ?it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  1%|          | 1/140 [00:06<14:38,  6.32s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  1%|▏         | 2/140 [00:06<06:49,  2.97s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  2%|▏         | 3/140 [00:07<04:15,  1.86s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  3%|▎         | 4/140 [00:08<03:10,  1.40s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  4%|▎         | 5/140 [00:08<02:09,  1.04it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  4%|▍         | 6/140 [00:08<01:48,  1.24it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  5%|▌         | 7/140 [00:09<01:41,  1.31it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  6%|▌         | 8/140 [00:10<01:32,  1.43it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  6%|▋         | 9/140 [00:10<01:28,  1.48it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  7%|▋         | 10/140 [00:11<01:21,  1.60it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  8%|▊         | 11/140 [00:11<01:23,  1.55it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  9%|▊         | 12/140 [00:12<01:19,  1.60it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "  9%|▉         | 13/140 [00:12<01:12,  1.76it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 10%|█         | 14/140 [00:13<01:11,  1.76it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 11%|█         | 15/140 [00:13<01:04,  1.93it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 11%|█▏        | 16/140 [00:14<00:58,  2.13it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 12%|█▏        | 17/140 [00:14<00:57,  2.14it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 13%|█▎        | 18/140 [00:15<00:55,  2.19it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 14%|█▎        | 19/140 [00:15<00:52,  2.28it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 14%|█▍        | 20/140 [00:16<01:03,  1.90it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 15%|█▌        | 21/140 [00:16<01:00,  1.95it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 16%|█▌        | 22/140 [00:17<00:55,  2.12it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 16%|█▋        | 23/140 [00:17<00:59,  1.96it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 17%|█▋        | 24/140 [00:18<00:57,  2.02it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 18%|█▊        | 25/140 [00:18<00:48,  2.36it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 19%|█▊        | 26/140 [00:18<00:44,  2.57it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 19%|█▉        | 27/140 [00:18<00:36,  3.14it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 20%|██        | 28/140 [00:19<00:54,  2.05it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 21%|██        | 29/140 [00:20<00:46,  2.39it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 21%|██▏       | 30/140 [00:20<00:40,  2.71it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 22%|██▏       | 31/140 [00:20<00:35,  3.08it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 23%|██▎       | 32/140 [00:20<00:37,  2.87it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 24%|██▎       | 33/140 [00:21<00:51,  2.07it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 24%|██▍       | 34/140 [00:22<00:53,  1.97it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 25%|██▌       | 35/140 [00:22<00:58,  1.80it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 26%|██▌       | 36/140 [00:23<01:10,  1.48it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 26%|██▋       | 37/140 [00:24<01:16,  1.34it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 27%|██▋       | 38/140 [00:25<01:07,  1.52it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 28%|██▊       | 39/140 [00:25<00:58,  1.74it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 29%|██▊       | 40/140 [00:26<00:51,  1.93it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 29%|██▉       | 41/140 [00:27<01:05,  1.50it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 30%|███       | 42/140 [00:27<01:00,  1.61it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 31%|███       | 43/140 [00:28<00:59,  1.62it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 31%|███▏      | 44/140 [00:28<00:55,  1.72it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 32%|███▏      | 45/140 [00:28<00:44,  2.13it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 33%|███▎      | 46/140 [00:29<00:39,  2.37it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 34%|███▎      | 47/140 [00:29<00:39,  2.36it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 34%|███▍      | 48/140 [00:30<00:39,  2.31it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 35%|███▌      | 49/140 [00:30<00:35,  2.56it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 36%|███▌      | 50/140 [00:30<00:39,  2.27it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 36%|███▋      | 51/140 [00:31<00:37,  2.37it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 37%|███▋      | 52/140 [00:32<00:45,  1.94it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 38%|███▊      | 53/140 [00:32<00:37,  2.30it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 39%|███▊      | 54/140 [00:32<00:38,  2.24it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 39%|███▉      | 55/140 [00:33<00:43,  1.94it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 40%|████      | 56/140 [00:34<00:43,  1.91it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 41%|████      | 57/140 [00:34<00:46,  1.78it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 41%|████▏     | 58/140 [00:35<01:01,  1.33it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 42%|████▏     | 59/140 [00:40<02:38,  1.95s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 43%|████▎     | 60/140 [00:41<02:20,  1.75s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 44%|████▎     | 61/140 [00:44<02:34,  1.96s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 44%|████▍     | 62/140 [00:45<02:04,  1.59s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 45%|████▌     | 63/140 [00:47<02:17,  1.79s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 46%|████▌     | 64/140 [00:47<01:39,  1.32s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 46%|████▋     | 65/140 [00:49<02:03,  1.64s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 47%|████▋     | 66/140 [00:50<01:30,  1.22s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 48%|████▊     | 67/140 [00:50<01:14,  1.03s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 49%|████▊     | 68/140 [00:51<01:01,  1.17it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 49%|████▉     | 69/140 [00:55<02:17,  1.94s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 50%|█████     | 70/140 [01:00<03:22,  2.89s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 51%|█████     | 71/140 [01:01<02:30,  2.19s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 51%|█████▏    | 72/140 [01:01<01:48,  1.60s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 52%|█████▏    | 73/140 [01:07<03:10,  2.85s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 53%|█████▎    | 74/140 [01:07<02:19,  2.11s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 54%|█████▎    | 75/140 [01:07<01:41,  1.56s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 54%|█████▍    | 76/140 [01:08<01:15,  1.18s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 55%|█████▌    | 77/140 [01:14<02:41,  2.56s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 56%|█████▌    | 78/140 [01:14<02:02,  1.98s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 56%|█████▋    | 79/140 [01:36<08:05,  7.95s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 57%|█████▋    | 80/140 [01:37<05:45,  5.77s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 58%|█████▊    | 81/140 [01:41<05:06,  5.20s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 59%|█████▊    | 82/140 [01:43<04:20,  4.49s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 59%|█████▉    | 83/140 [01:46<03:38,  3.84s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 60%|██████    | 84/140 [01:46<02:35,  2.78s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 61%|██████    | 85/140 [01:46<01:53,  2.07s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 61%|██████▏   | 86/140 [01:47<01:22,  1.52s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 62%|██████▏   | 87/140 [02:03<05:11,  5.87s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 63%|██████▎   | 88/140 [02:07<04:35,  5.29s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 64%|██████▎   | 89/140 [02:11<04:08,  4.87s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 64%|██████▍   | 90/140 [02:18<04:36,  5.52s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 65%|██████▌   | 91/140 [02:18<03:15,  3.99s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 66%|██████▌   | 92/140 [02:18<02:17,  2.86s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 66%|██████▋   | 93/140 [02:19<01:39,  2.11s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 67%|██████▋   | 94/140 [02:21<01:38,  2.13s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 68%|██████▊   | 95/140 [02:21<01:09,  1.55s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 69%|██████▊   | 96/140 [02:21<00:51,  1.16s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 69%|██████▉   | 97/140 [02:22<00:38,  1.12it/s]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 70%|███████   | 98/140 [02:45<05:18,  7.58s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 71%|███████   | 99/140 [02:45<03:39,  5.36s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 71%|███████▏  | 100/140 [02:56<04:40,  7.00s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 72%|███████▏  | 101/140 [03:01<04:07,  6.36s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 73%|███████▎  | 102/140 [03:01<02:51,  4.52s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 74%|███████▎  | 103/140 [03:01<01:59,  3.22s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 74%|███████▍  | 104/140 [03:01<01:24,  2.34s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 75%|███████▌  | 105/140 [03:20<04:08,  7.10s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 76%|███████▌  | 106/140 [03:28<04:12,  7.41s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 76%|███████▋  | 107/140 [03:28<02:59,  5.44s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 77%|███████▋  | 108/140 [03:29<02:03,  3.87s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 78%|███████▊  | 109/140 [03:29<01:26,  2.80s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 79%|███████▊  | 110/140 [04:16<08:04, 16.14s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 79%|███████▉  | 111/140 [04:31<07:37, 15.79s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 80%|████████  | 112/140 [04:49<07:36, 16.30s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 81%|████████  | 113/140 [04:57<06:15, 13.91s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 81%|████████▏ | 114/140 [04:57<04:16,  9.86s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 82%|████████▏ | 115/140 [04:59<03:06,  7.45s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 83%|████████▎ | 116/140 [05:15<04:01, 10.07s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 84%|████████▎ | 117/140 [05:16<02:43,  7.09s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 84%|████████▍ | 118/140 [05:16<01:51,  5.07s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 85%|████████▌ | 119/140 [05:21<01:45,  5.02s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 86%|████████▌ | 120/140 [05:21<01:11,  3.57s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 86%|████████▋ | 121/140 [05:59<04:21, 13.78s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 87%|████████▋ | 122/140 [06:02<03:12, 10.67s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 88%|████████▊ | 123/140 [06:06<02:27,  8.67s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 89%|████████▊ | 124/140 [06:09<01:49,  6.84s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 89%|████████▉ | 125/140 [06:10<01:19,  5.29s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 90%|█████████ | 126/140 [06:15<01:11,  5.10s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 91%|█████████ | 127/140 [06:16<00:48,  3.75s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 91%|█████████▏| 128/140 [06:19<00:43,  3.62s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 92%|█████████▏| 129/140 [06:20<00:32,  2.95s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 93%|█████████▎| 130/140 [06:23<00:28,  2.90s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 94%|█████████▎| 131/140 [06:24<00:21,  2.34s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 94%|█████████▍| 132/140 [06:27<00:19,  2.40s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 95%|█████████▌| 133/140 [06:28<00:15,  2.17s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 96%|█████████▌| 134/140 [06:31<00:13,  2.28s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 96%|█████████▋| 135/140 [06:34<00:12,  2.50s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 97%|█████████▋| 136/140 [06:34<00:07,  1.81s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 98%|█████████▊| 137/140 [06:35<00:04,  1.41s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 99%|█████████▊| 138/140 [06:40<00:05,  2.52s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      " 99%|█████████▉| 139/140 [06:45<00:03,  3.36s/it]\u001b[ATrying to set attribute `.obs` of view, copying.\n",
      "\n",
      "100%|██████████| 140/140 [06:45<00:00,  2.90s/it]\u001b[A\n"
     ]
    }
   ],
   "source": [
    "# split by sample id \n",
    "\n",
    "def flatten_list(t):\n",
    "    return [item for sublist in t for item in sublist]\n",
    "\n",
    "def create_raw_counts(sample_id):\n",
    "    df = pbmc_covid[pbmc_covid.obs.sampleID == sample_id]\n",
    "    sc.pp.filter_cells(df, min_genes=50) \n",
    "#     sc.pp.filter_genes(df, min_cells = 3) # avoid filtering genes, will need intersection of remaining genes, which filters to many out when subsequently filtering for LR pairs\n",
    "    return df\n",
    "\n",
    "sample_ids = list(set(flatten_list([sn.split('; ') for sn in Samples.sample_names.tolist()])))\n",
    "sample_counts = {sample_id: create_raw_counts(sample_id) for sample_id in tqdm(sample_ids)}\n",
    "\n",
    "\n",
    "min_cells_to_keep = min([df.n_obs for sample_id, df in sample_counts.items()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "decimal-mambo",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|          | 0/140 [00:00<?, ?it/s]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S054'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  1%|          | 1/140 [00:02<05:33,  2.40s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M038'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  1%|▏         | 2/140 [00:04<05:24,  2.35s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M064'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  2%|▏         | 3/140 [00:07<05:19,  2.33s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M014'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  3%|▎         | 4/140 [00:09<05:17,  2.33s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S035-4'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  4%|▎         | 5/140 [00:11<05:14,  2.33s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S026'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  4%|▍         | 6/140 [00:14<05:11,  2.33s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC001'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  5%|▌         | 7/140 [00:16<05:20,  2.41s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S056'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  6%|▌         | 8/140 [00:19<05:33,  2.52s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S024'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  6%|▋         | 9/140 [00:21<05:26,  2.49s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S044'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  7%|▋         | 10/140 [00:24<05:18,  2.45s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M049'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  8%|▊         | 11/140 [00:26<05:13,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S029'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  9%|▊         | 12/140 [00:28<05:08,  2.41s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M031-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      "  9%|▉         | 13/140 [00:31<05:03,  2.39s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M077'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 10%|█         | 14/140 [00:33<04:59,  2.38s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S083'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 11%|█         | 15/140 [00:35<04:56,  2.37s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S059'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 11%|█▏        | 16/140 [00:38<05:06,  2.47s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M008-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 12%|█▏        | 17/140 [00:41<05:09,  2.52s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M052'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 13%|█▎        | 18/140 [00:43<05:02,  2.48s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S068'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 14%|█▎        | 19/140 [00:46<04:57,  2.46s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S048'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      " 14%|█▍        | 20/140 [00:48<04:59,  2.50s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M073'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 15%|█▌        | 21/140 [00:51<04:53,  2.46s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S030'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 16%|█▌        | 22/140 [00:53<04:48,  2.44s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M053'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 16%|█▋        | 23/140 [00:55<04:44,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S078'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 17%|█▋        | 24/140 [00:58<04:41,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M058-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 18%|█▊        | 25/140 [01:00<04:38,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S090-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 19%|█▊        | 26/140 [01:03<04:35,  2.41s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S035-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 19%|█▉        | 27/140 [01:05<04:35,  2.44s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S014'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 20%|██        | 28/140 [01:08<04:41,  2.51s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S073-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 21%|██        | 29/140 [01:10<04:36,  2.49s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S023'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 21%|██▏       | 30/140 [01:13<04:31,  2.47s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC018-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 22%|██▏       | 31/140 [01:15<04:27,  2.45s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC021'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 23%|██▎       | 32/140 [01:18<04:27,  2.48s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M033'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 24%|██▎       | 33/140 [01:20<04:22,  2.45s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S055'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 24%|██▍       | 34/140 [01:22<04:19,  2.44s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S027'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 25%|██▌       | 35/140 [01:25<04:16,  2.44s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC002'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 26%|██▌       | 36/140 [01:27<04:13,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S037'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 26%|██▋       | 37/140 [01:30<04:09,  2.42s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC006'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 27%|██▋       | 38/140 [01:32<04:07,  2.42s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M027'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 28%|██▊       | 39/140 [01:35<04:11,  2.49s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M072'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      " 29%|██▊       | 40/140 [01:37<04:12,  2.52s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M034'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 29%|██▉       | 41/140 [01:40<04:10,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M037'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 30%|███       | 42/140 [01:42<04:04,  2.50s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S052'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 31%|███       | 43/140 [01:45<04:05,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC004'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 31%|███▏      | 44/140 [01:47<04:04,  2.54s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M007-6'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 32%|███▏      | 45/140 [01:50<03:58,  2.51s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M050'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 33%|███▎      | 46/140 [01:52<03:54,  2.49s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC023'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 34%|███▎      | 47/140 [01:55<03:50,  2.48s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M071'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 34%|███▍      | 48/140 [01:57<03:47,  2.47s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S025'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 35%|███▌      | 49/140 [02:00<03:44,  2.47s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC003'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 36%|███▌      | 50/140 [02:02<03:45,  2.50s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S021-4'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 36%|███▋      | 51/140 [02:05<03:44,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M030'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 37%|███▋      | 52/140 [02:07<03:42,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M059-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 38%|███▊      | 53/140 [02:10<03:38,  2.51s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S043'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 39%|███▊      | 54/140 [02:12<03:33,  2.49s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M048'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 39%|███▉      | 55/140 [02:15<03:33,  2.51s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S001-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 40%|████      | 56/140 [02:17<03:30,  2.51s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M044-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 41%|████      | 57/140 [02:20<03:27,  2.50s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC015'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 41%|████▏     | 58/140 [02:22<03:23,  2.49s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M012'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 42%|████▏     | 59/140 [02:25<03:21,  2.49s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M039-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      " 43%|████▎     | 60/140 [02:27<03:18,  2.49s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M074-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 44%|████▎     | 61/140 [02:30<03:18,  2.51s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S069-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 44%|████▍     | 62/140 [02:32<03:16,  2.52s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S060'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 45%|████▌     | 63/140 [02:35<03:16,  2.56s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S074-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 46%|████▌     | 64/140 [02:38<03:14,  2.56s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M022'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 46%|████▋     | 65/140 [02:40<03:10,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M009-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 47%|████▋     | 66/140 [02:43<03:08,  2.54s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M035-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 48%|████▊     | 67/140 [02:45<03:03,  2.52s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC009'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 49%|████▊     | 68/140 [02:48<03:00,  2.51s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M069'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 49%|████▉     | 69/140 [02:51<03:08,  2.66s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S013'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 50%|█████     | 70/140 [02:53<02:59,  2.57s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S041'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 51%|█████     | 71/140 [02:55<02:52,  2.50s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M028'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 51%|█████▏    | 72/140 [02:58<02:47,  2.46s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M004-6'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 52%|█████▏    | 73/140 [03:00<02:42,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S079'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 53%|█████▎    | 74/140 [03:02<02:40,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M013'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 54%|█████▎    | 75/140 [03:05<02:40,  2.47s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S033'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 54%|█████▍    | 76/140 [03:07<02:36,  2.44s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S047'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 55%|█████▌    | 77/140 [03:10<02:32,  2.42s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M054'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 56%|█████▌    | 78/140 [03:12<02:30,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S053'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 56%|█████▋    | 79/140 [03:15<02:27,  2.41s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC016'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      " 57%|█████▋    | 80/140 [03:17<02:24,  2.40s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M021'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 58%|█████▊    | 81/140 [03:19<02:22,  2.41s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M045'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 59%|█████▊    | 82/140 [03:23<02:32,  2.62s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M051'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 59%|█████▉    | 83/140 [03:28<03:11,  3.35s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC011'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 60%|██████    | 84/140 [03:33<03:35,  3.85s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S040'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 61%|██████    | 85/140 [03:37<03:42,  4.05s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M035-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 61%|██████▏   | 86/140 [03:39<03:11,  3.55s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC017'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 62%|██████▏   | 87/140 [03:42<02:51,  3.24s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC010'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 63%|██████▎   | 88/140 [03:44<02:36,  3.01s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC024'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 64%|██████▎   | 89/140 [03:47<02:23,  2.82s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M017'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 64%|██████▍   | 90/140 [03:49<02:14,  2.70s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M047'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 65%|██████▌   | 91/140 [03:52<02:06,  2.59s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S036-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 66%|██████▌   | 92/140 [03:54<02:01,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC008'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 66%|██████▋   | 93/140 [03:56<01:57,  2.49s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M041-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 67%|██████▋   | 94/140 [03:59<01:53,  2.47s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S061'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 68%|██████▊   | 95/140 [04:01<01:50,  2.45s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC019-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 69%|██████▊   | 96/140 [04:04<01:46,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M029'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 69%|██████▉   | 97/140 [04:06<01:45,  2.46s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M005'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 70%|███████   | 98/140 [04:09<01:44,  2.48s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S064'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 71%|███████   | 99/140 [04:11<01:41,  2.46s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S020'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      " 71%|███████▏  | 100/140 [04:13<01:37,  2.44s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M079'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 72%|███████▏  | 101/140 [04:16<01:34,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC020-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 73%|███████▎  | 102/140 [04:18<01:32,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC019-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 74%|███████▎  | 103/140 [04:21<01:29,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S034'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 74%|███████▍  | 104/140 [04:23<01:27,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M015'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 75%|███████▌  | 105/140 [04:26<01:24,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S089-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 76%|███████▌  | 106/140 [04:28<01:22,  2.43s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S092'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 76%|███████▋  | 107/140 [04:30<01:20,  2.44s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M026-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 77%|███████▋  | 108/140 [04:33<01:17,  2.41s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S081'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 78%|███████▊  | 109/140 [04:35<01:15,  2.45s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S018'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 79%|███████▊  | 110/140 [04:38<01:13,  2.47s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC014'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 79%|███████▉  | 111/140 [04:40<01:11,  2.47s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S019'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 80%|████████  | 112/140 [04:43<01:08,  2.46s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M006'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 81%|████████  | 113/140 [04:45<01:06,  2.46s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC018-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 81%|████████▏ | 114/140 [04:48<01:05,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M010-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 82%|████████▏ | 115/140 [04:50<01:03,  2.54s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M066'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 83%|████████▎ | 116/140 [04:53<01:00,  2.50s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M001'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 84%|████████▎ | 117/140 [04:55<00:56,  2.47s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC025'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 84%|████████▍ | 118/140 [04:58<00:54,  2.46s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M032'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 85%|████████▌ | 119/140 [05:00<00:51,  2.46s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S070-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      " 86%|████████▌ | 120/140 [05:03<00:48,  2.45s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S046'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 86%|████████▋ | 121/140 [05:05<00:46,  2.44s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M024'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 87%|████████▋ | 122/140 [05:08<00:45,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M018'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 88%|████████▊ | 123/140 [05:10<00:43,  2.57s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S049'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 89%|████████▊ | 124/140 [05:13<00:40,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC007'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 89%|████████▉ | 125/140 [05:16<00:38,  2.57s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC013'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 90%|█████████ | 126/140 [05:18<00:36,  2.58s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S015'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 91%|█████████ | 127/140 [05:21<00:33,  2.54s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S057'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 91%|█████████▏| 128/140 [05:23<00:30,  2.52s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M042-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 92%|█████████▏| 129/140 [05:26<00:27,  2.51s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S066'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 93%|█████████▎| 130/140 [05:28<00:25,  2.57s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S031'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 94%|█████████▎| 131/140 [05:31<00:23,  2.60s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M023'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 94%|█████████▍| 132/140 [05:33<00:20,  2.55s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S028'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 95%|█████████▌| 133/140 [05:36<00:17,  2.53s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M075'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 96%|█████████▌| 134/140 [05:38<00:15,  2.57s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC005'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 96%|█████████▋| 135/140 [05:41<00:12,  2.57s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M063-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 97%|█████████▋| 136/140 [05:44<00:10,  2.61s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC012'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 98%|█████████▊| 137/140 [05:46<00:07,  2.58s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M019'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 99%|█████████▊| 138/140 [05:49<00:05,  2.59s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S016'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "\n",
      " 99%|█████████▉| 139/140 [05:52<00:02,  2.62s/it]\u001b[A/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC022'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "100%|██████████| 140/140 [05:54<00:00,  2.53s/it]\u001b[A\n"
     ]
    }
   ],
   "source": [
    "# seed = 24\n",
    "seed += 1\n",
    "\n",
    "# subset to min cells_to_keep and write to csv\n",
    "cells_to_keep = list()\n",
    "for sample_id in tqdm(sample_counts):\n",
    "    random.seed(seed)\n",
    "    df = sample_counts[sample_id]\n",
    "    df = df[df.obs.index.isin(random.sample(df.obs.index.tolist(), min_cells_to_keep))] # subset\n",
    "#     df.to_df().to_csv(data_path + 'interim/umi_for_timing/' + sample_id + '.csv') # write\n",
    "    df.to_df().to_hdf(data_path + 'interim/timing_inputs/umi_per_sample.h5', key = sample_id)\n",
    "    cells_to_keep += df.obs.index.tolist()\n",
    "    seed += 1\n",
    "\n",
    "pbmc_covid.obs[pbmc_covid.obs.index.isin(cells_to_keep)].to_csv(data_path + 'interim/timing_inputs/metadata_for_timing.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "spectacular-release",
   "metadata": {},
   "source": [
    "# Classification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "completed-naples",
   "metadata": {},
   "outputs": [],
   "source": [
    "md = pd.read_excel(data_path + 'raw/GSE158055_sample_metadata.xlsx', sheet_name = 0, skiprows=20)\n",
    "md = md.iloc[range(304 - 20), range(25)]\n",
    "md = md[md['Sample name'].isin(samples_to_keep)] # still filter for datasets with > 2000 cells"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "opposed-retirement",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_ids = md['Sample name'].unique().tolist()\n",
    "pbmc_covid.obs[pbmc_covid.obs.sampleID.isin(sample_ids)].to_csv(data_path + 'interim/classification_inputs/metadata.csv')\n",
    "sample_counts = {sample_id: create_raw_counts(sample_id) for sample_id in tqdm(sample_ids)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "delayed-question",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  2%|▏         | 4/227 [00:17<18:54,  5.09s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M043-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  2%|▏         | 5/227 [00:25<21:38,  5.85s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M048'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  3%|▎         | 6/227 [00:32<22:51,  6.20s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M044-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  3%|▎         | 7/227 [00:38<23:12,  6.33s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M043-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  4%|▎         | 8/227 [00:45<24:09,  6.62s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S054'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  4%|▍         | 9/227 [00:52<24:15,  6.68s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S056'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  4%|▍         | 10/227 [00:59<24:10,  6.68s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M042-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  5%|▍         | 11/227 [01:04<22:27,  6.24s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M041-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  5%|▌         | 12/227 [01:10<21:23,  5.97s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M049'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  6%|▌         | 13/227 [01:15<20:42,  5.81s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M046'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  6%|▌         | 14/227 [01:20<19:50,  5.59s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M047'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  7%|▋         | 15/227 [01:25<19:01,  5.38s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S055'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  7%|▋         | 16/227 [01:31<19:33,  5.56s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S057'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  7%|▋         | 17/227 [01:36<19:14,  5.50s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M045'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  8%|▊         | 18/227 [01:42<19:55,  5.72s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M041-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  8%|▊         | 19/227 [01:49<20:16,  5.85s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M042-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  9%|▉         | 20/227 [01:56<21:43,  6.30s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC008'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "  9%|▉         | 21/227 [02:00<19:32,  5.69s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC009'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 10%|▉         | 22/227 [02:05<18:42,  5.47s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC010'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 10%|█         | 23/227 [02:11<19:05,  5.62s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC011'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 11%|█         | 24/227 [02:16<18:12,  5.38s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC012'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 11%|█         | 25/227 [02:22<18:27,  5.48s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M055'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 11%|█▏        | 26/227 [02:27<17:51,  5.33s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M053'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 12%|█▏        | 27/227 [02:34<19:18,  5.79s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S067'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 12%|█▏        | 28/227 [02:40<19:44,  5.95s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S065'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 13%|█▎        | 29/227 [02:49<23:00,  6.97s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M051'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 13%|█▎        | 30/227 [02:55<21:57,  6.69s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC013'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 14%|█▎        | 31/227 [03:07<26:57,  8.25s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC014'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 14%|█▍        | 32/227 [03:17<27:59,  8.61s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC015'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 15%|█▍        | 33/227 [03:29<31:08,  9.63s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC016'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 15%|█▍        | 34/227 [03:37<29:23,  9.14s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC017'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 15%|█▌        | 35/227 [03:48<31:19,  9.79s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S064'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 16%|█▌        | 36/227 [03:52<25:17,  7.94s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M054'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 16%|█▋        | 37/227 [03:59<24:21,  7.69s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M052'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 17%|█▋        | 38/227 [04:04<22:26,  7.12s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S068'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 17%|█▋        | 39/227 [04:10<21:14,  6.78s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S066'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 18%|█▊        | 40/227 [04:18<21:52,  7.02s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S059'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 18%|█▊        | 41/227 [04:22<19:17,  6.22s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S060'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 19%|█▊        | 42/227 [04:28<18:43,  6.07s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M050'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 19%|█▉        | 43/227 [04:33<17:08,  5.59s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S061'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 19%|█▉        | 44/227 [04:36<15:02,  4.93s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S062'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 20%|█▉        | 45/227 [04:39<13:33,  4.47s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S063'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 20%|██        | 46/227 [04:43<12:46,  4.23s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M061-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 21%|██        | 47/227 [04:47<12:09,  4.06s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M061-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 21%|██        | 48/227 [04:56<16:52,  5.66s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC018-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 22%|██▏       | 49/227 [04:59<14:25,  4.86s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC018-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 22%|██▏       | 50/227 [05:04<14:02,  4.76s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S073-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 22%|██▏       | 51/227 [05:06<12:16,  4.18s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S074-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 23%|██▎       | 52/227 [05:09<10:35,  3.63s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M058-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 23%|██▎       | 53/227 [05:13<11:07,  3.83s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M063-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 24%|██▍       | 54/227 [05:16<10:23,  3.61s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC019-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 24%|██▍       | 55/227 [05:20<10:07,  3.53s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC019-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 25%|██▍       | 56/227 [05:22<09:30,  3.33s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M059-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 25%|██▌       | 57/227 [05:27<10:09,  3.58s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M059-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 26%|██▌       | 58/227 [05:31<11:05,  3.94s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M060-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 26%|██▌       | 59/227 [05:34<10:03,  3.59s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M060-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 26%|██▋       | 60/227 [05:37<09:29,  3.41s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S076-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 27%|██▋       | 61/227 [05:41<09:56,  3.60s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S076-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 27%|██▋       | 62/227 [05:44<09:01,  3.28s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC020-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 28%|██▊       | 63/227 [05:47<09:18,  3.41s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC020-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 28%|██▊       | 64/227 [05:52<10:15,  3.78s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S035-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*`; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 29%|██▊       | 65/227 [05:54<09:08,  3.38s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S035-4'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 29%|██▉       | 66/227 [05:58<08:54,  3.32s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S036-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 30%|██▉       | 67/227 [06:01<08:59,  3.37s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M064'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 30%|██▉       | 68/227 [06:12<14:30,  5.48s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M066'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 30%|███       | 69/227 [06:24<19:52,  7.54s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M067'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 31%|███       | 70/227 [06:36<23:30,  8.99s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M068'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 31%|███▏      | 71/227 [06:39<18:23,  7.07s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M074-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 32%|███▏      | 72/227 [06:43<15:38,  6.05s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M075'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 32%|███▏      | 73/227 [06:46<13:36,  5.30s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S086-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 33%|███▎      | 74/227 [07:03<22:39,  8.88s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S085-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 33%|███▎      | 75/227 [07:18<26:51, 10.60s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S090-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 33%|███▎      | 76/227 [07:21<20:52,  8.29s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S090-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 34%|███▍      | 77/227 [07:25<17:30,  7.00s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S092'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 34%|███▍      | 78/227 [07:36<20:16,  8.16s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M076-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 35%|███▍      | 79/227 [07:49<23:36,  9.57s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M074-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 35%|███▌      | 80/227 [08:00<24:46, 10.11s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S088-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 36%|███▌      | 81/227 [08:12<26:10, 10.75s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S085-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 36%|███▌      | 82/227 [08:26<28:20, 11.73s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S089-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 37%|███▋      | 83/227 [08:40<29:32, 12.31s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S087-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 37%|███▋      | 84/227 [08:49<26:51, 11.27s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S086-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 37%|███▋      | 85/227 [09:03<28:51, 12.19s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M077'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 38%|███▊      | 86/227 [09:11<25:19, 10.78s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M078'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 38%|███▊      | 87/227 [09:20<24:19, 10.43s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M079'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 39%|███▉      | 88/227 [09:29<23:07,  9.98s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC001'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 39%|███▉      | 89/227 [09:39<22:51,  9.94s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC002'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 40%|███▉      | 90/227 [09:51<24:10, 10.59s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S024'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 40%|████      | 91/227 [09:59<21:54,  9.66s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S026'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 41%|████      | 92/227 [10:08<21:52,  9.72s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M013'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 41%|████      | 93/227 [10:11<17:15,  7.73s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M014'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 41%|████▏     | 94/227 [10:21<18:31,  8.36s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M015'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 42%|████▏     | 95/227 [10:32<19:52,  9.04s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M016'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 42%|████▏     | 96/227 [10:40<19:25,  8.89s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S023'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 43%|████▎     | 97/227 [10:44<15:49,  7.30s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S025'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 43%|████▎     | 98/227 [10:48<13:40,  6.36s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S027'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 44%|████▎     | 99/227 [10:55<14:04,  6.60s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S034'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 44%|████▍     | 100/227 [11:00<12:28,  5.90s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M025'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 44%|████▍     | 101/227 [11:04<11:10,  5.32s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S033'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 45%|████▍     | 102/227 [11:09<11:03,  5.30s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC003'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 45%|████▌     | 103/227 [11:15<11:19,  5.48s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC004'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 46%|████▌     | 104/227 [11:22<12:15,  5.98s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M028'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 46%|████▋     | 105/227 [11:25<10:30,  5.17s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M029'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 47%|████▋     | 106/227 [11:29<09:51,  4.89s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M027'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 47%|████▋     | 107/227 [11:35<09:56,  4.97s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M026-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 48%|████▊     | 108/227 [11:39<09:43,  4.90s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S032-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 48%|████▊     | 109/227 [11:43<08:54,  4.53s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M026-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 48%|████▊     | 110/227 [11:47<08:43,  4.47s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M026-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 49%|████▉     | 111/227 [11:51<08:16,  4.28s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M004-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 49%|████▉     | 112/227 [11:59<10:10,  5.30s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M004-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 50%|████▉     | 113/227 [12:11<14:13,  7.49s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M004-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 50%|█████     | 114/227 [12:21<15:07,  8.03s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S022-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 51%|█████     | 115/227 [12:27<13:58,  7.49s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S022-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 51%|█████     | 116/227 [12:32<12:28,  6.74s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S022-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 52%|█████▏    | 117/227 [12:40<12:48,  6.99s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S022-4'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 52%|█████▏    | 118/227 [12:47<12:44,  7.01s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S022-5'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 52%|█████▏    | 119/227 [12:55<13:06,  7.29s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M010-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 53%|█████▎    | 120/227 [13:01<12:39,  7.10s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M010-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 53%|█████▎    | 121/227 [13:07<11:51,  6.72s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M010-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 54%|█████▎    | 122/227 [13:10<10:02,  5.73s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M010-4'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 54%|█████▍    | 123/227 [13:17<10:22,  5.98s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M010-5'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 55%|█████▍    | 124/227 [13:24<10:49,  6.30s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M010-6'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*`; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 55%|█████▌    | 125/227 [13:31<10:56,  6.44s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M009-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 56%|█████▌    | 126/227 [13:39<11:52,  7.06s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M011'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 56%|█████▌    | 127/227 [13:48<12:41,  7.62s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M012'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 56%|█████▋    | 128/227 [13:56<12:48,  7.76s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M005'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 57%|█████▋    | 129/227 [14:05<12:55,  7.91s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M006'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 57%|█████▋    | 130/227 [14:11<11:56,  7.38s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M004-4'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 58%|█████▊    | 131/227 [14:19<12:08,  7.58s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M004-5'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 58%|█████▊    | 132/227 [14:29<13:20,  8.43s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M004-6'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 59%|█████▊    | 133/227 [14:37<12:56,  8.26s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S013'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 59%|█████▉    | 134/227 [14:42<11:16,  7.28s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S014'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 59%|█████▉    | 135/227 [14:48<10:41,  6.97s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S015'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 60%|█████▉    | 136/227 [14:53<09:20,  6.16s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S016'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 60%|██████    | 137/227 [15:00<09:37,  6.41s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S017'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 61%|██████    | 138/227 [15:06<09:29,  6.39s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S018'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 61%|██████    | 139/227 [15:14<09:57,  6.79s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S019'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 62%|██████▏   | 140/227 [15:20<09:44,  6.72s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S020'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 62%|██████▏   | 141/227 [15:29<10:23,  7.25s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M007-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 63%|██████▎   | 142/227 [15:36<10:10,  7.18s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M007-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 63%|██████▎   | 143/227 [15:44<10:39,  7.61s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M007-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 63%|██████▎   | 144/227 [15:52<10:23,  7.51s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M007-4'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*`; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 64%|██████▍   | 145/227 [15:56<08:55,  6.53s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M007-5'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 64%|██████▍   | 146/227 [16:03<08:57,  6.64s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M007-6'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 65%|██████▍   | 147/227 [16:06<07:28,  5.60s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M008-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 65%|██████▌   | 148/227 [16:13<07:53,  6.00s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M008-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 66%|██████▌   | 149/227 [16:22<08:56,  6.87s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M009-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 66%|██████▌   | 150/227 [16:26<07:51,  6.12s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M009-4'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 67%|██████▋   | 151/227 [16:31<07:08,  5.64s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M009-5'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 67%|██████▋   | 152/227 [16:35<06:34,  5.26s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M009-6'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 67%|██████▋   | 153/227 [16:40<06:18,  5.12s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S021-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 68%|██████▊   | 154/227 [16:46<06:43,  5.53s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S021-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 68%|██████▊   | 155/227 [16:53<07:05,  5.90s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S021-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 69%|██████▊   | 156/227 [17:00<07:19,  6.20s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S021-4'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 69%|██████▉   | 157/227 [17:05<06:48,  5.83s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S021-5'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 70%|██████▉   | 158/227 [17:10<06:25,  5.59s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M018'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 70%|███████   | 159/227 [17:17<06:54,  6.09s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S029'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 70%|███████   | 160/227 [17:25<07:24,  6.64s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S030'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 71%|███████   | 161/227 [17:31<07:05,  6.45s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M023'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 71%|███████▏  | 162/227 [17:39<07:26,  6.87s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S031'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 72%|███████▏  | 163/227 [17:46<07:19,  6.87s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M024'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 72%|███████▏  | 164/227 [17:52<07:07,  6.78s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M017'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*`; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 73%|███████▎  | 165/227 [18:01<07:39,  7.42s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M019'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 73%|███████▎  | 166/227 [18:10<07:50,  7.71s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M020'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 74%|███████▎  | 167/227 [18:19<08:05,  8.10s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M021'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 74%|███████▍  | 168/227 [18:27<08:05,  8.24s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S028'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 74%|███████▍  | 169/227 [18:35<07:53,  8.16s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M022'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 75%|███████▍  | 170/227 [18:43<07:34,  7.98s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M001'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 75%|███████▌  | 171/227 [18:46<06:07,  6.56s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M030'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 76%|███████▌  | 172/227 [18:56<07:00,  7.64s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M031-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 76%|███████▌  | 173/227 [19:06<07:25,  8.25s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M031-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 77%|███████▋  | 174/227 [19:14<07:18,  8.28s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M032'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 77%|███████▋  | 175/227 [19:23<07:19,  8.45s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M033'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 78%|███████▊  | 176/227 [19:36<08:23,  9.88s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M034'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 78%|███████▊  | 177/227 [19:50<09:09, 10.98s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M035-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 78%|███████▊  | 178/227 [20:00<08:41, 10.64s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M035-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 79%|███████▉  | 179/227 [20:04<07:02,  8.79s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M036-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 79%|███████▉  | 180/227 [20:18<07:58, 10.18s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M036-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 80%|███████▉  | 181/227 [20:35<09:32, 12.45s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M037'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 80%|████████  | 182/227 [20:45<08:46, 11.70s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M038'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 81%|████████  | 183/227 [20:55<08:04, 11.00s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M039-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 81%|████████  | 184/227 [21:07<08:16, 11.54s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M039-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 81%|████████▏ | 185/227 [21:23<08:55, 12.74s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M040-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 82%|████████▏ | 186/227 [21:39<09:25, 13.79s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M040-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 82%|████████▏ | 187/227 [21:54<09:21, 14.03s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC005'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 83%|████████▎ | 188/227 [22:04<08:23, 12.92s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC006'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 83%|████████▎ | 189/227 [22:11<07:02, 11.12s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC007'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 84%|████████▎ | 190/227 [22:19<06:19, 10.26s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S012-1'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 84%|████████▍ | 191/227 [22:23<05:02,  8.41s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S001-2'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 85%|████████▍ | 192/227 [22:28<04:10,  7.16s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S001-3'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 85%|████████▌ | 193/227 [22:35<04:08,  7.31s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC021'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 85%|████████▌ | 194/227 [22:42<03:56,  7.18s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC022'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 86%|████████▌ | 195/227 [22:48<03:36,  6.78s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC023'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 86%|████████▋ | 196/227 [22:54<03:19,  6.43s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC024'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 87%|████████▋ | 197/227 [23:00<03:10,  6.35s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-HC025'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 87%|████████▋ | 198/227 [23:05<02:50,  5.87s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M069'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 88%|████████▊ | 199/227 [23:10<02:45,  5.90s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M070'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 88%|████████▊ | 200/227 [23:16<02:37,  5.82s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M071'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 89%|████████▊ | 201/227 [23:22<02:28,  5.73s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M072'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 89%|████████▉ | 202/227 [23:27<02:18,  5.55s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-M073'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 89%|████████▉ | 203/227 [23:35<02:28,  6.20s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S077'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 90%|████████▉ | 204/227 [23:41<02:21,  6.15s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S078'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*`; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 90%|█████████ | 205/227 [23:47<02:19,  6.35s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S079'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 91%|█████████ | 206/227 [23:53<02:08,  6.12s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S080'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 91%|█████████ | 207/227 [23:57<01:52,  5.65s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S081'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 92%|█████████▏| 208/227 [24:01<01:37,  5.15s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S082'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 92%|█████████▏| 209/227 [24:09<01:46,  5.89s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S083'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 93%|█████████▎| 210/227 [24:15<01:42,  6.01s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S084'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 93%|█████████▎| 211/227 [24:24<01:47,  6.74s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S037'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 93%|█████████▎| 212/227 [24:36<02:05,  8.40s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S038'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 94%|█████████▍| 213/227 [24:49<02:17,  9.79s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S040'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 94%|█████████▍| 214/227 [24:55<01:53,  8.72s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S041'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 95%|█████████▍| 215/227 [25:04<01:44,  8.74s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S042'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 95%|█████████▌| 216/227 [25:12<01:33,  8.54s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S043'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 96%|█████████▌| 217/227 [25:20<01:22,  8.27s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S044'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 96%|█████████▌| 218/227 [25:28<01:14,  8.24s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S045'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 96%|█████████▋| 219/227 [25:33<00:57,  7.21s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S046'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 97%|█████████▋| 220/227 [25:42<00:55,  7.93s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S047'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 97%|█████████▋| 221/227 [25:55<00:56,  9.38s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S048'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 98%|█████████▊| 222/227 [26:13<00:59, 11.85s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S049'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 98%|█████████▊| 223/227 [26:25<00:47, 11.84s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S050'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 99%|█████████▊| 224/227 [26:43<00:41, 13.69s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S051'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      " 99%|█████████▉| 225/227 [26:50<00:23, 11.75s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S052'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "100%|█████████▉| 226/227 [26:58<00:10, 10.77s/it]/home/hratch/anaconda3/envs/cellchat/lib/python3.9/site-packages/tables/path.py:155: NaturalNameWarning: object name is not a valid Python identifier: 'S-S053'; it does not match the pattern ``^[a-zA-Z_][a-zA-Z0-9_]*$``; you will not be able to use natural naming to access this object; using ``getattr()`` will still work, though\n",
      "  check_attribute_name(name)\n",
      "100%|██████████| 227/227 [27:18<00:00,  7.22s/it]\n"
     ]
    }
   ],
   "source": [
    "for sample_id in tqdm(sample_counts):\n",
    "    df = sample_counts[sample_id]\n",
    "#     df.to_df().to_csv(data_path + 'interim/umi_for_classification/' + sample_id + '.csv') # write\n",
    "    df.to_df().to_hdf(data_path + 'interim/classification_inputs/umi_per_sample.h5', key = sample_id)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:cellchat] *",
   "language": "python",
   "name": "conda-env-cellchat-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
